getline.ts/src/common.ts

```diff
@@ -92,6 +96,72 @@ export class Token extends Address {
         let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
         return token.call<void>('approve', spender.ascii, value);
     }
+
+    /**
+     * Returns this token's symbol.
+     * @returns Token symbol.
+     */
+    public async symbol(): Promise<string> {
```

Comment:
All three accessors could be replaced by a private function similar to this, less code duplication, no need to declare caching (also mostly reusable in any Contract):

```typescript
private cache: { [key: string]: any } = {};

public async symbol() = constantFromContract<string>('symbol')

private async constantFromContract<T>(what: string) {
    if (this.cache[what] != undefined) {
        return this[what];
    }
    let token = await this.blockchain.existing(TOKEN_CONTRACT, this);
    this.cache[what] = await token.call<T>(what);
    return this.cache[what];
}
```

Reply:
Done.